src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLRequestSerializer.java

```diff
@@ -37,8 +38,7 @@ public static String serialize(GraphQLRequest graphQLRequest) {
             builder.append(graphQLRequest.getResponseProjection().toString());
```

Comment:
Do we expect this `toString()` here to properly encode string literals respecting the GraphQL specification?
http://spec.graphql.org/June2018/#sec-Appendix-Grammar-Summary.Lexical-Tokens

I think this requires quite some work to be done properly. We should probably build an actual query data model (or use a potentially internal one from `graphql-java`) defining all the elements that constitute a GraphQL query/mutation like a query field, nested field, input parameters, directives etc.

Reply:
@joffrey-bion That's a very valid point. I didn't pay attention to it.
I've refactored the whole serialization logic. Could you please take a look again?